config/prefix.go

```diff
@@ -28,21 +28,18 @@ import (
 	gogoproto "github.com/gogo/protobuf/proto"
 )
 
-// PrefixConfig relate a string prefix to a config object. Config
-// objects include accounting, permissions, users, and zones. PrefixConfig
-// objects are the constituents of PrefixConfigMap objects. In order
-// to support binary searches of hierarchical prefixes (see the
-// comments in NewPrefixConfigMap), PrefixConfig objects are
-// additionally added to a PrefixConfigMap to demarcate the end of a
-// prefix range. Such end-of-range sentinels need to refer back to the
-// next "higher-up" prefix in the hierarchy (many times this is the
-// default prefix which covers the entire range of keys). The Canonical
-// key refers to this "higher-up" PrefixConfig by specifying its prefix
-// so it can be binary searched from within a PrefixConfigMap.
-type PrefixConfig struct {
-	Prefix    proto.Key         // the prefix the config affects
-	Canonical proto.Key         // the prefix for the canonical config, if applicable
-	Config    gogoproto.Message // the config object
+// MakePrefixConfig creates a new PrefixConfig.
+func MakePrefixConfig(prefix, canonical proto.Key, configI gogoproto.Message) PrefixConfig {
```

Comment:
Is this function needed?  Can you just create a new proto.PrefixConfig directly?  Or is the configUnion the issue?